### PR TITLE
Add admin add n delete product

### PR DESCRIPTION
**Pull request by imranhossainemi**
### Description

- [ ] Add Admin page component with demo product list to project from dev branch.

#### General

- [ ] The code follows the project's coding guidelines and best practices.
- [ ] The code is modular, well-organized, and easy to understand.
- [ ] Variable and function names are meaningful and self-descriptive.
- [ ] There are  commented-out code blocks or unused variables/methods for api declaration for future use

#### Functionality

- [ ] The changes accomplish the intended purpose and work as expected.
- [ ] The existing functionality is not affected adversely by the changes.
- [ ] All edge cases and possible error scenarios have been considered.

#### Testing

- [ ] The changes include appropriate unit tests for new functionality have not been tested.
- [ ] Test for responsive design

### CODE REVIEW

File: src/pages/AdminPage.jsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+
+    if (!productPrice.trim()) {
+      isValid = false;
+      errors.price = 'Product price is required';
+    } else if (isNaN(productPrice) || +productPrice <= 0) {
+      isValid = false;
+      errors.price = 'Product price must be a positive number';
+    }
```
Comment (shasherazi) [inline src/pages/AdminPage.jsx:62]: I would suggest that instead of manually checking the presence of every value, just use the `required` attribute for the corresponding inputs in the form.

File: src/pages/AdminPage.jsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            />
+            {formErrors.name && <span className="error">{formErrors.name}</span>}
+          </div>
+          <div className='from-fields'>
+            <label>Product Price:</label>
+            <input
+              type="text"
+              value={productPrice}
+              onChange={(e) => setProductPrice(e.target.value)}
+            />
```
Comment (shasherazi) [inline src/pages/AdminPage.jsx:137]: Price should be a number

File: src/pages/AdminPage.jsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+              onChange={(e) => setProductMemory(e.target.value)}
+            />
+            {formErrors.memory && <span className="error">{formErrors.memory}</span>}
+          </div>
+          <div className='from-fields'>
+            <label>Product Storage:</label>
+            <input
+              type="text"
+              value={productStorage}
+              onChange={(e) => setProductStorage(e.target.value)}
```
Comment (shasherazi) [inline src/pages/AdminPage.jsx:172]: Memory and storage should also be numbers

File: src/pages/AdminPage.jsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+                }
+                if (product.name.toLowerCase().includes(searchKeyword.toLowerCase())) {
+                  return product;
+                }
+                return null;
+              }).map((product) => (
+                <li key={product.id}>
+                  <p>{product.name}</p>
+                  <button onClick={() => handleDeleteProduct(product.id)}>Delete</button>
+                </li>
```
Comment (shasherazi) [inline src/pages/AdminPage.jsx:203]: These delete buttons are not aligned when product names are of different length.

File: src/pages/AdminPage.jsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+                if (searchKeyword === '') {
+                  return product;
+                }
+                if (product.name.toLowerCase().includes(searchKeyword.toLowerCase())) {
+                  return product;
+                }
+                return null;
+              }).map((product) => (
+                <li key={product.id}>
+                  <p>{product.name}</p>
```
Comment (shasherazi) [inline src/pages/AdminPage.jsx:201]: I think that the product name should be a link to the Product show page.

**Review by shasherazi — CHANGES_REQUESTED** [review]
Hello @imranhossainemi, good effort. However, I think there'r room for improvement. See the comments below.

File: src/pages/AdminPage.jsx (hunk 4 above)
Comment (imranhossainemi) [reply src/pages/AdminPage.jsx:196]: @shasherazi What! Can I have screenshot what are seeing?

File: src/pages/AdminPage.jsx (hunk 1 above)
Comment (imranhossainemi) [reply src/pages/AdminPage.jsx:62]: We try this for positive number validation

File: src/pages/AdminPage.jsx (hunk 4 above)
Comment (shasherazi) [reply src/pages/AdminPage.jsx:196]: image

File: src/pages/AdminPage.jsx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    if (!productMemory.trim()) {
+      isValid = false;
+      errors.memory = 'Product memory is required to be Nmbers only';
+    } else if (isNaN(productMemory) || +productMemory <= 0) {
+      isValid = false;
+      errors.memory = 'Product memory must be a positive number';
+    }
+    if (!productStorage.trim()) {
+      isValid = false;
+      errors.storage = 'Product productStorage is required to be Nmbers only';
```
Comment (fuadmoin) [inline src/pages/AdminPage.jsx:63]: There is a typo, `Product productStorage`, please fix that.

**Review by fuadmoin — APPROVED** [review]
Hi @imranhossainemi   ,  Well done on the changes you made suggested by @shasherazi .

## Highlights!
- Descriptive pull request ✔️ 
- No linter errors ✔️ 
- Professional Gitflow ✔️ 



Cheers, and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

-[OPTIONAL] Please fix the typo on the comment down below.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------

File: src/pages/AdminPage.jsx (hunk 6 above)
Comment (imranhossainemi) [reply src/pages/AdminPage.jsx:63]: Thank you
